### DISCUSSION TITLE
No Debug informations just SMTP connect() failed

### DISCUSSION THREAD

**Discussion by Firehold** [PHPMailer/PHPMailer#2949] [Q&A] [answered]
Hello,

despite the debug level (SMTP::DEBUG_LOWLEVEL / 4), which should show me all messages, I only get the error message:
`Mailer Error: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting`
E-Mail credential data was tested with Thunderbird before, they're correct.

Can you tell me how to get more information about the error?

**PHPMailer-Version:** 6.8.0
**PHP-Version:** 8.1
**openssl Extension:** loaded

**Code:**
```php
<?php
#####################
#    LOAD MAILER    #
#####################

    require BASEPATH . '/includes/PHPMailer-master/src/Exception.php';
    require BASEPATH . '/includes/PHPMailer-master/src/PHPMailer.php';
    require BASEPATH . '/includes/PHPMailer-master/src/SMTP.php';

    use PHPMailer\PHPMailer\PHPMailer;
    use PHPMailer\PHPMailer\SMTP;
    use PHPMailer\PHPMailer\Exception;

    $phpmail_customer_mail = new PHPMailer();
    
    $phpmail_customer_mail->isSMTP();
    $phpmail_customer_mail->Host       = SMTP_HOST;
    $phpmail_customer_mail->SMTPAuth   = true;
    $phpmail_customer_mail->Username   = SMTP_USER;
    $phpmail_customer_mail->Password   = SMTP_PASS;
    $phpmail_customer_mail->SMTPSecure = SMTP_PROTOCOL; 
    $phpmail_customer_mail->Port       = SMTP_PORT;
    $phpmail_customer_mail->setFrom(SMTP_MAIL, $from_name);
    $phpmail_customer_mail->addReplyTo(SMTP_MAIL, $from_name);
    // Set Debug Level
    $phpmail_mail->SMTPDebug = SMTP::DEBUG_LOWLEVEL;
    // Add Receiver
    $phpmail_customer_mail->addAddress($mailto, $sender);
    // Add File
    $phpmail_customer_mail->addAttachment($file);

    //Content
    $phpmail_customer_mail->Subject = $subject;
    $phpmail_customer_mail->msgHTML($mailtext);
    $phpmail_customer_mail->AltBody = $mailtext;
    $phpmail_customer_mail->CharSet = 'UTF-8';

    // Send Mail
    if (!$phpmail_customer_mail->send()) {
        echo 'Mailer Error: ' . $phpmail_customer_mail->ErrorInfo;
    } else {
        echo 'Message sent!';
    }   
}
```

**Comment by Firehold** [comment]
After changing the debug-level to the number 4 i got more informations, (although not much more):
```
2023-08-27 13:40:28 Connection: opening to myserver.com:465, timeout=300, options=array()
2023-08-27 13:40:28 Connection: opened
2023-08-27 13:45:28 SMTP -> get_lines(): select timed-out in (300 sec)
2023-08-27 13:45:28 SERVER -> CLIENT:
2023-08-27 13:45:28 Connection: closing due to error
2023-08-27 13:45:28 Connection: closed
SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting
Mailer Error: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting 
```
Is there a possibility to get more informations about the `closing due to error` ?

**Reply by Synchro** [reply]
That's quite a bit of useful information. Servers will immediately start a TLS handshake on port 465 – but if `SMTPSecure` is not set to `ssl`, it will fail like this, so check the value you are using in your `SMTP_PROTOCOL` constant. It may also behave this way if traffic is being blocked - though it looks like that is not the case because the TCP connection does succeed in opening. The 300 second wait is part of the SMTP spec, but if your server is nearby, you can expect this to be pretty instant.

**Reply by Firehold** [reply]
Thank you very much for your quick response.
My constants for the Mailsettings are like this:

```php
define('SMTP_HOST',     'myserver.com');
define('SMTP_PORT',     '465');
define('SMTP_USER',     'm123456');
define('SMTP_PASS',     '**************');
define('SMTP_MAIL',     'mail@myserver.com');
define('SMTP_PROTOCOL', 'SSL');
```
Is it possible, that this error occurs from a bad SSL-certificate for the domain the script is called from? Because of its on a staging site currently the SSL certificate is not valid. But i don't know if the http-encryption has an effect on the PHPMail-Script.

The time to get this errors is roughly 3 Minutes.

**Reply by Synchro** [reply, marked as answer]
Port is an int:

    define('SMTP_PORT',     465);

`SMTPSecure` is case-sensitive:

    define('SMTP_PROTOCOL', 'ssl');

It also has its own constant already: `PHPMailer::ENCRYPTION_SMTPS`

**Reply by Firehold** [reply]
Oh my god, it was litterally just the case sensitive SSL ...
Thank you so much for the quick and professional help especially on a Sunday ❤️
